demo/project_demo10-FraudDetectionDeltaLake/run.py

```diff
@@ -0,0 +1,331 @@
+
+    # Start the pipeline to continuously process the input stream of credit card
+    # transactions and output newly computed feature vectors to a Delta table.
+
+    sql.start()
+
+def build_program(client, pipeline_name):
+    sql = SQLContext(pipeline_name, client).get_or_create()
+    # Declare input table with raw credit card transaction data.
+    sql.register_table(
+        "TRANSACTION",
+        SQLSchema(
+            {
+                "trans_date_trans_time": "TIMESTAMP",
+                "cc_num": "BIGINT",
+                "merchant": "STRING",
+                "category": "STRING",
+                "amt": "DOUBLE",
+                "trans_num": "STRING",
+                "unix_time": "BIGINT",
+                "merch_lat": "DOUBLE",
+                "merch_long": "DOUBLE",
+                "is_fraud": "BIGINT",
+            }
+        ),
+    )
+
+    # Declare input table with demographics data.
+    sql.register_table(
+        "DEMOGRAPHICS",
+        SQLSchema(
+            {
+                "cc_num": "BIGINT",
+                "first": "STRING",
+                "last": "STRING",
+                "gender": "STRING",
+                "street": "STRING",
+                "city": "STRING",
+                "state": "STRING",
+                "zip": "BIGINT",
+                "lat": "DOUBLE",
+                "long": "DOUBLE",
+                "city_pop": "BIGINT",
+                "job": "STRING",
+                "dob": "DATE",
+            }
+        ),
+    )
+
+    # Feature query.
+    query = """
+        SELECT
+           t.cc_num,
+           dayofweek(trans_date_trans_time) as d,
+           CASE
+             WHEN dayofweek(trans_date_trans_time) IN(6, 7) THEN true
+             ELSE false
+           END AS is_weekend,
+           hour(trans_date_trans_time) as hour_of_day,
+           CASE
+             WHEN hour(trans_date_trans_time) <= 6 THEN true
+             ELSE false
+           END AS is_night,
+           -- Average spending per day, per week, and per month.
+           AVG(amt) OVER window_1_day AS avg_spend_pd,
+           AVG(amt) OVER window_7_day AS avg_spend_pw,
+           AVG(amt) OVER window_30_day AS avg_spend_pm,
+           -- Average spending over the last three months for the same day of the week.
+           COALESCE(
+            AVG(amt) OVER (
+              PARTITION BY t.cc_num, EXTRACT(DAY FROM trans_date_trans_time)
+              ORDER BY unix_time
+              RANGE BETWEEN 7776000 PRECEDING and CURRENT ROW
+            ), 0) AS avg_spend_p3m_over_d,
+           -- Number of transactions in the last 24 hours.
+           COUNT(*) OVER window_1_day AS trans_freq_24,
+           amt, unix_time, zip, city_pop, is_fraud
+        FROM transaction as t
+        JOIN demographics as d
+        ON t.cc_num = d.cc_num
+        WINDOW
+          window_1_day AS (PARTITION BY t.cc_num ORDER BY unix_time RANGE BETWEEN 86400 PRECEDING AND CURRENT ROW),
```

Comment:
have to fix this to compute in dates
doesn't it work with the latest PRs for partitioned_rolling_aggregates?